astroquery/linelists/cdms/core.py

```diff
@@ -170,6 +172,13 @@ def query_lines_async(self, min_frequency, max_frequency, *,
         response2 = self._request(method='GET', url=fullurl,
                                   timeout=self.TIMEOUT, cache=cache)
 
+        # accounts for three formats, e.g.: '058501' or 'H2C2S' or '058501 H2C2S'
+        badlist = (self.MALFORMATTED_MOLECULE_LIST +  # noqa
+                   [y for x in self.MALFORMATTED_MOLECULE_LIST for y in x.split()])
+        if payload['Molecules'] in badlist:
```

Comment:
I would say drop line 146, as request can deal with a dict data input, but you cannot index it if you turn in to a list of tuples. A dictionary would be anyways better as it would be consistent with our docstrings (even though upstream accepts more types right now).